### PR TITLE
Add front end testing to NoRent confirmation page, preview page, and log out page

### PR DESCRIPTION
**Pull request by sraby**
This PR adds some front end testing to the pages of our NoRent site! Specifically, I added tests to the confirmation page, preview page, and log out page, which were all identified by CodeClimate to be lacking sufficient testing.

### CODE REVIEW

File: frontend/lib/norent/letter-builder/tests/confirmation.test.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        }),
+        norentLatestLetter: trackingNumber
+          ? {
+              trackingNumber: trackingNumber,
+              letterSentAt: null,
+              paymentDate: "Boop 1st, 2099",
+            }
+          : null,
+      },
+    });
```
Comment (toolness) [inline frontend/lib/norent/letter-builder/tests/confirmation.test.tsx:26]: Nice, just FYI i started a "session builder" class to make it easier to construct session objects, especially across different tests.  it's totally optional to use but there if you need it, feel free to add methods to it and stuff if you want: https://github.com/JustFixNYC/tenants2/blob/master/frontend/lib/tests/session-builder.tsx

You can see an example of it in use here:

https://github.com/JustFixNYC/tenants2/pull/1331/files#diff-482a964ea6464f8d2e446ac81e289e44

Totally up to you if you want to use it or not tho!

File: frontend/lib/norent/letter-builder/tests/confirmation.test.tsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  };
+
+  it("should work", () => {
+    const pal = createPal("NY", "1234Boopy");
+    pal.rr.getByText("You've sent your letter");
+    expect(pal.rr.getByText("Sign the petition").getAttribute("href")).toBe(
+      CANCEL_RENT_PETITION_URL
+    );
```
Comment (toolness) [inline frontend/lib/norent/letter-builder/tests/confirmation.test.tsx:34]: In #1429 I've added a helper function to `AppTesterPal` called `ensureLinkGoesTo` which should make this less verbose!

File: frontend/lib/norent/letter-builder/tests/confirmation.test.tsx (hunk 1 above)
Comment (sraby) [reply frontend/lib/norent/letter-builder/tests/confirmation.test.tsx:26]: Ok cool! Ok I'll check it out

File: frontend/lib/norent/letter-builder/tests/confirmation.test.tsx (hunk 2 above)
Comment (sraby) [reply frontend/lib/norent/letter-builder/tests/confirmation.test.tsx]: Nice, just updated this.

**Comment by sraby** [comment]
@toolness Apologies if we already discussed this— but is there anything that I need to add to these tests (like checking the Send Letter mutation) or should I go ahead and merge them?

**Pull request merged**